src/frontend/js/components/CourseAddToWishlist/index.logout.spec.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Test suite for CourseAddToWishlist component
+ * for anonymous visitors
+ */
+import { render, screen, waitFor } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+import { IntlProvider } from 'react-intl';
+import { QueryClientProvider } from '@tanstack/react-query';
+import userEvent from '@testing-library/user-event';
+import * as mockFactories from 'utils/test/factories';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import JoanieApiProvider from 'data/JoanieApiProvider';
+import { SessionProvider } from 'data/SessionProvider';
+import { location } from 'utils/indirection/window';
+import CourseAddToWishlist from '.';
+
+jest.mock('utils/indirection/window', () => ({
+  location: {
+    pathname: '/tests/CourseAddToWishlist/',
+    assign: jest.fn(() => true),
+  },
+}));
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: {
+        backend: 'fonzie',
+        endpoint: 'https://authentication.test',
+      },
+      joanie_backend: {
+        endpoint: 'https://joanie.test',
+      },
+    })
+    .generate(),
+}));
+
+const renderCourseAddToWishlist = (courseCode: string) =>
+  render(
+    <IntlProvider locale="en">
+      <QueryClientProvider client={createTestQueryClient({ user: false })}>
```

Comment:
You can `null` as value to hydrate query state with an anonymous user. In this way you do not have to mock the authentication request.
```suggestion
      <QueryClientProvider client={createTestQueryClient({ user: null })}>
```